core/src/main/kotlin/Utilities/DownloadSamples.kt

```diff
@@ -0,0 +1,64 @@
+package org.jetbrains.dokka.Utilities
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import java.io.File
+import java.io.FileOutputStream
+
+object DownloadSamples {
+
+    val client = OkHttpClient()
+
+    fun downloadSamples(): Boolean {
+
+        //loop through each directory of AOSP code in SamplesPathsToURLs.kt
+        filepathsToUrls.forEach { (filepath, url) ->
+
+            //build request using each URL
+            val request = Request.Builder()
+                .url(url)
+                .build()
+            val response = client.newCall(request).execute()
+
+            if (response.isSuccessful) {
+
+                //save .tar.gz file to filepath designated by map
+                var currentFile = File(filepath)
+                currentFile.mkdirs()
+
+                var fos = FileOutputStream(filepath + ".tar.gz")
+                fos.write(response.body?.bytes())
+                fos.close()
+
+                //Unzip, Untar, and delete compressed file after
+                extractFiles(filepath)
+
+            } else {
+                println("Download of samples failed.")
```

Comment:
Does the `response` object provide an error message?  If so, may be helpful to display the error here